Assignment_05.rb

```diff
@@ -0,0 +1,151 @@
+                                        [DateTime.parse("2023-01-01 14:30:00"), DateTime.parse("2023-01-01 15:30:00")]] },
+    { id: 4, name: "task4", time_logs: [[DateTime.parse("2023-01-01 15:30:00"), DateTime.parse("2023-01-01 16:30:00")]] },
+    { id: 5, name: "task5", time_logs: [[DateTime.parse("2023-01-01 16:30:00"), DateTime.parse("2023-01-01 17:30:00")]] },
+    { id: 6, name: "task6", time_logs: [[DateTime.parse("2023-01-01 17:30:00"), DateTime.parse("2023-01-01 18:30:00")]] },
+  ]
+
+
+expected output format:
+
+  {
+    categories: ["task1", "task2", "task3", "task4", "task5", "task6"],
+    total_time: [<total_spent_time_in_minutes_for_task1>,  .. <total_spent_time_in_minutes_for_task6>],
+    total_time: [<(total_spent_time_in_minutes_for_task_1 / (24 * 60)) * 100>, .. <(total_spent_time_in_minutes_for_task_6 / (24 * 60)) * 100>]
+  }
+
+  
+1. Define required class & methods for it
+2. split down the main method into multiple methods meaningfully
+3. validate input & handle exception for invalid input
+4. write unit test cases for this class functionalities using rspec
+=end
+
+require 'date'
+
+class Task
+  attr_accessor :id, :name, :time_logs
+
+  def initialize(id="deafult", name="deafult", time_logs=[])
+    @id = id
+    @name = name
+    @time_logs = time_logs
+  end
+
+  # This method returns the total time used for the task
+  def total_time_in_minute
+    time_logs.sum { |time_log| get_time_duration_in_minute(time_log[0], time_log[1]) }.round
+  end
+
+  # This method takes total time used for a task and returns the in percentage form for a day
+  def occupation_percecntage_day_wise(total_time_in_minute)
+    (total_time_in_minute.to_f / (24 * 60) * 100).round(2)
+  end
+
+  # This method does the validation
+  def valid?
+    validate_time_logs
+  end
+
+  private
+  
+  # This methods takes an time log array as argument [start_time, end_time] and retures the duration in minute
+  def get_time_duration_in_minute(start_time, end_time)
+    duration = DateTime.parse(end_time) - DateTime.parse(start_time)
+    duration.to_f * 24 * 60
+  end
+
+  # This method validate the the time logs
+  def validate_time_logs
+    time_logs.each do |time_log|
+      if DateTime.parse(time_log[0]) > DateTime.parse(time_log[1])
+        puts "Time logs validation failed - end time earlier than start time - for id = #{id} and name = #{name}"
+        return false
+      elsif DateTime.parse(time_log[0]) == DateTime.parse(time_log[1])
+        puts "Time logs validation failed - start time and end time can't be same - for id = #{id} and name = #{name}"
+        return false
+      end
+    end
+    true
+  end
+end
+
+class TaskHandler
+  attr_reader :tasks
+
+  def initialize
+    @tasks = []
+  end
+
+  def add_one_task(task_attributes)
+    task = Task.new(task_attributes[:id], task_attributes[:name], task_attributes[:time_logs])
+    if task.valid?
```

Comment:
please give line break when you are starting a block, and no indentation is there.